src/zkevm_specs/evm/execution/error_invalid_opcode.py

```diff
@@ -0,0 +1,74 @@
+from ...util import FQ
+from ..execution_state import ExecutionState
+from ..instruction import Instruction, Transition
+from ..table import CallContextFieldTag
+
+SEPARATE_INVALID_OPCODES = [
+    0x0C,
+    0x0D,
+    0x0E,
+    0x0F,
+    0x1E,
+    0x1F,
+    0x5C,
+    0x5D,
+    0x5E,
+    0x5F,
+    0xF6,
+    0xF7,
+    0xF8,
+    0xF9,
+    0xFB,
+    0xFC,
+    0xFE,
+]
+
+# Gadget for invalid opcodes. It verifies invalid bytes in any condition of:
+# - `opcode > 0x20 && opcode < 0x30`
+# - `opcode > 0x48 && opcode < 0x50`
+# - `opcode > 0xA4 && opcode < 0xF0`
+# - one of [`SEPARATE_INVALID_OPCODES`]
+def invalid_opcode(instruction: Instruction):
+    # Fixed lookup for invalid opcode.
+    opcode = instruction.opcode_lookup(True)
+    instruction.responsible_opcode_lookup(opcode)
+
+    op_gt_20, _ = instruction.compare(FQ(0x20), opcode, 1)
```

Comment:
After some investigation, I found that `FixedTableTag.InvalidOpcode` was deleted in PR https://github.com/privacy-scaling-explorations/zkevm-specs/pull/110 and @han0110 commented with https://github.com/privacy-scaling-explorations/zkevm-specs/pull/110#discussion_r814976468.

I am sorry that I could not confirm if this `5 compares + one set constrain` is correct. Or I should use fourth column `auxiliary` of `ResponsibleOpcode` to specify it's an invalid opcode. WDYT?

Also @han0110 may have some suggestions. Thanks.

Reply:
> I'm wondering if is not cheaper just to do a lookup that 5 compares + one constrain per `SEPARATE_INVALID_OPCODES`

if I remember correct, one lookup generate 4 gates 😄

Reply:
Sorry I have missed this, so I think the `responsible_opcode_lookup` already does the opcode check, since only the invalid opcode could pass the lookup, or perhaps I'm missing something?

The invalid opcode responsible opcodes are generated here:

https://github.com/privacy-scaling-explorations/zkevm-specs/blob/13a5cf983deea12dcf2972242d3d876a9a0337dd/src/zkevm_specs/evm/execution_state.py#L354-L355

Reply:
Yes. I see. I am sorry for my previous misunderstanding. It seems that [responsible_opcode_lookup](https://github.com/privacy-scaling-explorations/zkevm-specs/blob/master/src/zkevm_specs/evm/instruction.py#L708) is enough, it has already lookup for the invalid opcode with `ExecutionState.ErrorInvalidOpcode`.

And it seems that I should also map the [invalid_opcodes](https://github.com/privacy-scaling-explorations/zkevm-specs/blob/d971d8e9dd4b8146affd5c5df438b838a672c3a6/src/zkevm_specs/evm/opcode.py#L358) function to `ExecutionState::ErrorInvalidOpcode` in [responsible_opcodes](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/zkevm-circuits/src/evm_circuit/step.rs#L172) of zkevm-circuit. So I could do the same lookup in circuit. Thanks.

Reply:
Ah you are right! That's currently missed in circuit side.

Reply:
Thanks. Will update the code.

Reply:
Fixed to only do fixed lookup for `ResponsibleOpcode` in 721a1ca.
And also update the circuit PR https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1089